### DISCUSSION TITLE
API ideas

### DISCUSSION THREAD

**Discussion by the1mason** [pterodactyl/panel#4258] [Feature Requests]
My little suggestion is about API.
When someone sends a request to the API with an incorrect token server returns 200 status code with HTML body.
I think it could be better to receive 401 status code with JSON error, empty body, or the same HTML page but with unauthorized code. It would be much more obvious and easier to handle.

Edit:
Also, getting non-existent file contents/file list returns 200 status code and HTML, so my second idea is to use 400/404 status code or an error.

**Comment by devnote-dev** [comment]
@the1mason HTML is returned if you don't specify the `Content-Type` and `Accept` headers in your request, the panel assumes that a browser or similar is accessing the endpoint and responds accordingly. For JSON responses with the correct status code you need to set these headers to "application/json".
